### PR TITLE
Admin UI: Encountering infinite loading when accessing pages while logged out

### PR DESCRIPTION
**Pull request by stamenione**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart [hunk 1]
```diff
@@ -45,18 +45,22 @@ class _SplashScreenState extends State<SplashScreen> {
     await GetIt.I.allReady();
     final sp = await SharedPreferences.getInstance();
 
-    if (!sp.containsKey('api_key')) return _navigate('/login');
+    if (!sp.containsKey('api_key')) return _navigateToLogin();
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:48]: after login the page you wanted to go is completely lost, right?

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart (hunk 1 above)
Comment (stamenione) [reply Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:48]: I see what you mean.

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart [hunk 2]
```diff
@@ -45,18 +45,23 @@ class _SplashScreenState extends State<SplashScreen> {
...
-    if (!isValid) return _navigate('/login');
+    if (!isValid) return _navigateToLogin();
 
     GetIt.I.registerSingleton(await AdminApiClient.create(baseUrl: baseUrl, apiKey: apiKey));
     return _navigate('/identities');
   }
 
+  void _navigateToLogin() {
+    final redirectPath = widget.redirect != null ? Uri.encodeComponent(widget.redirect!) : '/';
+    context.go('/login?redirect=$redirectPath');
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:62]: no.

1. `widget.redirect` is already encoded
2. / as default?! login has a default and we don't want to override it

```suggestion
    if (widget.redirect != null) context.go('/login?redirect=${widget.redirect!}');

    context.go('/login');
```

Comment (jkoenig134) [reply Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:62]: or even easier

```suggestion
    context.go(widget.redirect != null ? '/login?redirect=${widget.redirect!}' : '/login');
```

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart [hunk 3]
```diff
@@ -45,18 +45,23 @@ class _SplashScreenState extends State<SplashScreen> {
...
+  }
+
   void _navigate(String defaultRoute) {
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:65]: this navigate method is now only used one. I don't see a reason why this still exists.

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart [hunk 4]
```diff
@@ -54,18 +54,10 @@ class _SplashScreenState extends State<SplashScreen> {
     if (!isValid) return _navigateToLogin();
 
     GetIt.I.registerSingleton(await AdminApiClient.create(baseUrl: baseUrl, apiKey: apiKey));
-    return _navigate('/identities');
+    if (mounted) context.go(widget.redirect != null ? Uri.decodeComponent(widget.redirect!) : '/identities');
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:57]: early return..

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart [hunk 5]
```diff
@@ -45,23 +45,19 @@ class _SplashScreenState extends State<SplashScreen> {
...
   }
 
-  void _navigate(String defaultRoute) {
-    if (!mounted) return;
-
-    if (widget.redirect != null) return context.go(Uri.decodeComponent(widget.redirect!));
-
-    context.go(defaultRoute);
+  void _navigateToLogin() {
+    context.go(widget.redirect != null ? '/login?redirect=${widget.redirect!}' : '/login');
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:61]: why isn't this an arrow function?

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart (hunk 4 above)
Comment (jkoenig134) [reply Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:57]: and I liked this more like it was in the method, in more lines.

File: Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart [hunk 6]
```diff
@@ -45,23 +45,22 @@ class _SplashScreenState extends State<SplashScreen> {
...
-    return _navigate('/identities');
-  }
 
-  void _navigate(String defaultRoute) {
     if (!mounted) return;
 
     if (widget.redirect != null) return context.go(Uri.decodeComponent(widget.redirect!));
 
-    context.go(defaultRoute);
+    return context.go('/identities');
```
Comment (jkoenig134) [inline Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart:62]: I don't want to annoy you .. but your only job here was to copy paste the methods content to place where the method was called.

Where does this return come from now?!

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**